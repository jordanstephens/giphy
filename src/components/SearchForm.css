.search-form {
  flex-grow: 1;
  text-align: right;
  display: flex;
  margin-left: 40px;
}

@media (max-width: 500px) {
  .search-form { margin-left: 0; width: 100%; }
}

.search-field {
  flex-grow: 1;
  font-size: 16px;
  padding: 10px;
  border: none;
}

.search-button {
  background: rgb(0, 204, 255);
  border: none;
  color: #fff;
  padding: 10px;
  cursor: pointer;
  font-size: 16px;
  font-weight: bold;
}