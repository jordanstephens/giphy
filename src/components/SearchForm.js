import React from 'react';
import PropTypes from 'prop-types';

import './SearchForm.css';

class SearchForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      q: this.props.q || ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onSubmit(event) {
    const { q } = this.state;
    this.props.onSubmit(event, { q });
  }

  onChange(event) {
    const q = event.target.value;
    this.setState({ q });
  }

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input type="text" name="q" value={this.state.q} onChange={this.onChange} />
        <button type="submit">Search</button>
      </form>
    );
  }
}

SearchForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  q: PropTypes.string
};

export default SearchForm;
