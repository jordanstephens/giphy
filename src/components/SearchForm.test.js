/*global expect*/
import React from 'react';
import SearchForm from './SearchForm';
import renderer from 'react-test-renderer';

it('renders without q', () => {
  const tree = renderer
    .create(<SearchForm onSubmit={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with q', () => {
  const tree = renderer
    .create(<SearchForm q='test' onSubmit={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
