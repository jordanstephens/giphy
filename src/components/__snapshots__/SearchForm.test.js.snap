// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with q 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    className="search-field"
    name="q"
    onChange={[Function]}
    placeholder="Search GIFs"
    type="text"
    value="test"
  />
  <button
    className="search-button"
    type="submit"
  >
    Go!
  </button>
</form>
`;

exports[`renders without q 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    className="search-field"
    name="q"
    onChange={[Function]}
    placeholder="Search GIFs"
    type="text"
    value=""
  />
  <button
    className="search-button"
    type="submit"
  >
    Go!
  </button>
</form>
`;
